flash-patcher.py

```diff
@@ -24,111 +15,147 @@
 
-    # windows default install location
-    if (set_jpexs_if_exists(f"{os.getenv('ProgramFiles')}\\FFDec\\ffdec.exe")):
-        return True
+ARGS_FLATPAK = [
+    "run",
+    "--branch=stable",
+    "--arch=x86_64",
+    "--command=ffdec.sh",
+    "com.jpexs.decompiler.flash",
+]
 
-    # wow64 install location
-    if (set_jpexs_if_exists(f"{os.getenv('ProgramFiles(x86)')}\\FFDec\\ffdec.exe")):
-        return True
 
-    return False
+class JPEXSInterface:
+    """An interface to interact with JPEXS via the shell."""
 
-def perror(mesg):
-    print(mesg, file=sys.stderr)
+    path: Path
+    args: list
 
-def read_from_file(file_location, patch_file, current_line_no):
-    try:
-        with open(file_location) as f:
-            current_file = f.readlines()
-            return current_file
-    except (FileNotFoundError, IsADirectoryError) as e:
-        perror("")
-        perror(patch_file + ", line " + str(current_line_no) + ": Invalid injection location")
-        perror("Could not find or load SWF decompiled file at: " + file_location)
-        perror("Aborting...")
-        exit(1)
-
-def write_to_file(path, lines):
-    with open(path, "w") as f:
-        f.writelines(lines)
+    def __init__(self: JPEXSInterface, path: Path, args: list | None = None) -> None:
+        self.path = path
+        if args is None:
+            self.args = []
+        else:
+            self.args = args
+
+    def dump_xml(
+        self: JPEXSInterface,
+        inputfile: Path,
+        output_dir: Path,
+    ) -> subprocess.CompletedProcess:
+        """Dump XML data of the input file into the output location."""
+        return subprocess.run(
+            [self.path, *self.args, "-swf2xml", inputfile, output_dir],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            check=True,
+        )
+
+    def rebuild_xml(
+        self: JPEXSInterface,
+        input_dir: Path,
+        output_file: Path,
+    ) -> subprocess.CompletedProcess:
+        """Rebuild XML data from a directory into an output SWF file."""
+        return subprocess.run(
+            [self.path, *self.args, "-xml2swf", input_dir, output_file],
+            stdout=subprocess.DEVNULL,
+            check=True,
+        )
+
+    def export_scripts(
+        self: JPEXSInterface,
+        inputfile: Path,
+        output_dir: Path,
+    ) -> subprocess.CompletedProcess:
+        """Export scripts from a SWF file into a directory."""
+        info("Exporting scripts into %s...", output_dir)
+        return subprocess.run(
+            [
+                self.path,
+                *self.args,
+                "-export",
+                "script",
+                output_dir,
+                inputfile,
+            ],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            check=True,
+        )
+
+    def recompile_data(
+        self: JPEXSInterface,
+        part: str,
```

Comment:
add a comment with all valid values of `part`

Reply:
Good thinking.